Client/core/CGUI.cpp

```diff
@@ -431,7 +425,12 @@ void CLocalGUI::SetMainMenuVisible(bool bVisible)
         }
 
         if (!bVisible)
-            pGUI->SetCursorAlpha(pGUI->GetCurrentServerCursorAlpha());
+        {
+            float r, g, b, a; 
+            pGUI->GetCursorColor(r, g, b, a); //restore the cursor color when we close the main menu
+            pGUI->SetCursorColor(r, g, b, a);
+        }
+           
```

Comment:
whats the point of this code?

Reply:
when we close the main menu it will retrieve the cursor color and set it again

Reply:
Why do we set it again? Wouldnt that change nothing?

Reply:
it just part of a some code in Client\core\CMessageLoopHook.cpp
I think some sort of keybind when we press escape 
```
        // Pass escape keyup to onClientKey
        if (uMsg == WM_KEYUP && wParam == VK_ESCAPE)
        {
            g_pCore->GetKeyBinds()->TriggerKeyStrokeHandler("escape", uMsg == WM_KEYDOWN, true);
            return true;
        }

        // Suppress auto repeat of escape and console toggle keys
        if ((uMsg == WM_KEYDOWN || uMsg == WM_CHAR) && (wParam == VK_ESCAPE || wParam == VK_F8 || wParam == '`'))
        {
            bool bFirstHit = (lParam & 0x40000000) ? false : true;
            if (!bFirstHit)
                return true;
        }

        // Slightly hacky way of suppressing escape character when console is closed with escape key
        if (uMsg == WM_CHAR && wParam == VK_ESCAPE)
        {
            bool bTemp = ms_bIgnoreNextEscapeCharacter;
            ms_bIgnoreNextEscapeCharacter = false;
            if (bTemp)
                return true;
        }

        if (CKeyBinds::IsFakeCtrl_L(uMsg, wParam, lParam))
            return true;

        bool       bWasCaptureKey = false;
        CMainMenu* pMainMenu = g_pCore->GetLocalGUI()->GetMainMenu();
        if (pMainMenu)
        {
            CSettings* pSettings = pMainMenu->GetSettingsWindow();
            if (pSettings)
            {
                if (uMsg == WM_KEYDOWN && wParam == VK_ESCAPE && GetJoystickManager()->IsCapturingAxis())
                {
                    GetJoystickManager()->CancelCaptureAxis(true);
                    return true;
                }
                bWasCaptureKey = (pSettings->IsCapturingKey() && pSettings->ProcessMessage(uMsg, wParam, lParam));
                if (!bWasCaptureKey)
                {
                    // If Escape is pressed and we're playing ingame, we show/hide the menu
                    if (uMsg == WM_KEYDOWN && wParam == VK_ESCAPE && g_pCore->IsConnected())
                    {
                        // Hide the console
                        CConsoleInterface* pConsole = g_pCore->GetConsole();
                        if (pConsole->IsVisible())
                        {
                            ms_bIgnoreNextEscapeCharacter = true;
                            pConsole->SetVisible(false);

                            CGUI* pGUI = g_pCore->GetGUI();
                            float r, g, b, alpha;

                            if (!g_pCore->IsMenuVisible())
                            {
                                pGUI->GetCursorColor(r, g, b, alpha);
                                pGUI->SetCursorColor(r, g, b, alpha);
                            }
                      
                            return true;
                        }

                        // The mainmenu makes sure it isn't hidden if UseIngameButtons == false
                        if (!CCore::GetSingleton().IsOfflineMod())
                        {
                            if (g_pCore->GetKeyBinds()->TriggerKeyStrokeHandler("escape", uMsg == WM_KEYDOWN, true))
                            {
                                // Stop chat input
                                if (CLocalGUI::GetSingleton().IsChatBoxInputEnabled())
                                {
                                    CLocalGUI::GetSingleton().SetChatBoxInputEnabled(false);
                                    return true;
                                }

                                CLocalGUI::GetSingleton().SetMainMenuVisible(!CLocalGUI::GetSingleton().IsMainMenuVisible());
                            }
                            return true;
                        }
                    }
                    else if (uMsg == WM_KEYDOWN && wParam == VK_ESCAPE && !g_pCore->IsConnected())
                    {
                        // If Escape is pressed and we're not playing ingame, hide certain windows
                        CLocalGUI::GetSingleton().GetMainMenu()->OnEscapePressedOffLine();
                    }
```

Reply:
> if you open the main menu the cursor color will be white if we close the menu again it will return the color again
> 
> it just part of a some code in Client\core\CMessageLoopHook.cpp I think some sort of keybind handler when we press escape so if we press escape to close the menu it will call `void CLocalGUI::SetMainMenuVisible(bool bVisible)` then we get to the setCursorColor

I mean, `GetCursorColor` will return current color. Doing `SetCursorColor` with the same values doesnt make any sense.
If without that the code works correctly then something is horribly wrong behind the scene

Reply:
fixed in
(https://github.com/multitheftauto/mtasa-blue/pull/3558/commits/127efaa1b0192a7c3f14754b55d4789df742c4f1)